CM.Middleware/GlobalErrorHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.AspNetCore.Http;
+using System.Globalization;
+using System.Net;
+using System.Text.Json;
+
+namespace CM.Middleware
+{
+    public class GlobalErrorHandler
+    {
+        private readonly RequestDelegate _next;
+
+        public GlobalErrorHandler(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (Exception error)
+            {
+                var response = context.Response;
+                response.ContentType = "application/json";
+
+                switch (error)
+                {
+                    case CustomException e:
+                        response.StatusCode = (int)HttpStatusCode.BadRequest;
+                        break;
+                    case KeyNotFoundException e:
+                        response.StatusCode = (int)HttpStatusCode.NotFound;
+                        break;
+                    default:
+                        response.StatusCode = (int)HttpStatusCode.InternalServerError;
+                        break;
+                }
+
+                var result = JsonSerializer.Serialize(new { message = error?.Message });
```

Comment:
Try not to send the error msg stack trace for 500 errors in release mode but okay for debug mode. Try to implement this using conditional preprocessor directives.

Reply:
error?.Message is not referring to the whole stack trace as I know.  And it's a bit confusing for me, about the preprocessor directives usage here.

Reply:
Added custom error messages.